### ISSUE TITLE
Error in Bowtie2 "isn't numeric in bitwise and (&) at "

### ISSUE THREAD

**Issue by hasnahana** [BenLangmead/bowtie2#141]
Hi All and @BenLangmead ,
 
I am trying to align and then remove rrna sequences from my RNA-Seq data.
I built index for my rrna fasta files using the following command
**bowtie2-build -f --threads 24 --quiet rrna.fasta index_rrna**

Now, i am trying to align my paired-end fastq files to the rrna files using the following command-
**bowtie2 --very-sensitive-local --phred33  -x index_rrna -1 file_R1.fastq  -2 file_R2.fastq --threads 24 --met-file summary_rrna.txt --al-conc-gz paired_matches_file --un-conc-gz paired_unmatched_file --al-gz single_matches_file --un-gz single_unmatched_file**

However, on running the script with slurm, my job gets killed and I am getting multiple errors of similar kind. See below-
****Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42321883.
Argument "\0:1214:7890:64386 2:N:0:TGACCA%0ACATCGCATTCTACATTATTTGC..." isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42323448.
Argument "DDDDDDDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42333056.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42339084.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42351116.
Argument "\0BCC%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42360088.
Argument "%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42367614.
Argument "CDBDDDDDDDDDDDDC%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42387194.
Argument "%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42397563.
Argument "\0JIIIJIIJIJIJJJIJIJJJJFGIJJJJJJIJJJJJJJJJJJIJJHHHHFDFFF..." isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42400464.
Argument "%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42401981.
Argument "FFFFFEEEEEDDDDDDDEDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42403454.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42403463.
Argument "\0CAB?@<C%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42407951.
Argument "EACD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42411008.
Argument "\0FFECCEEEDDDDDDDDDDDDDDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42411013.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42426317.
Argument "\0GACAAGTTTCAACATATACTTCGTATTATGAATACGAATA%0A+%0AHB?E>FF..." isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42429058.
Argument "\0DDDDDDDDBDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42430603.
Argument "%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42447041.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42454466.
Argument "\0DDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42462019.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42463580.
Argument "\0CC<>@BBBB>>:>CB%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42469559.
Argument "\0CCC%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42475460.
Argument "\0EDEDDDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42475477.
Argument "\0FFDECCE%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42479966.
Argument "\0A%0AAAACACAAATAAAAAACTTTGGCGTCCTTTCTCTATACAATTTGATATTT..." isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42487379.
Argument "DDCD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42545351.
Argument "\0BCD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42556600.
Argument "DDDDDDDDDCCDDDDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42601991.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42606424.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42613837.
Argument "%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42620040.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42620093.
Argument "\0DDA%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42623154.
Argument "\0TTCATGGCATCTACACTACGGTTATTCCTAGTTCTAGTAAATTCTATGTAGCCA..." isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42626621.
Argument "DDDDDDDD%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42632202.
Argument "\00" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42636822.
Argument "%0" isn't numeric in bitwise and (&) at  bowtie2-2.3.3/bowtie2 line 556, <BT> line 42659081.
slurmstepd: *** JOB 141619 ON cn069 CANCELLED AT 2017-09-16T18:09:48 ***

Please help.

**Comment by ch4rr0** [comment]
Would it be possible to share your input files so that I can speedily recreate this issue?

**Comment by mkoohim** [comment]
I guess you have a problem in your input files. Please check your input file to find if you have any space or undesirable characters in it.

**Comment by BenLangmead** [comment]
Please post a small example where this happens, and please also share the contents of the `bowtie2` script.  At first glance, that error message seems to indicate your `bowtie2` script differs from the version released in 2.3.3.

**Comment by nick-youngblut** [comment]
I keep getting the same thing when using bowtie2 as part of the [humann2 pipeline](http://huttenhower.sph.harvard.edu/humann2). For instance, humann2 will map illumina reads (paired-end combined into one file) against a reference database of markers: 

`bowtie2 --quiet --no-unal --very-sensitive -S out.sam -mpa_v20_m200 -p 30 -U R1-R2_trim_filt.fq`

The output is often something like the following:

```
Argument "\00" isn't numeric in bitwise and (&) at /ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/ef9ee16c/bin/bowtie2 line 562, <BT> line 1416056.
Argument "\00" isn't numeric in bitwise and (&) at /ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/ef9ee16c/bin/bowtie2 line 562, <BT> line 1424833.
```

The errors will occur randomly during runs. Re-running the exact same job will often work the 2nd time (or 3rd or 4th). This only seems to happen when I run humann2 (and thus bowtie2) on a SGE compute cluster running Ubuntu 16.04.3. At first I thought it was some buffering issue with humann2 (it pipes reads in and pipes the bowtie2 sam output out with subprocess), but even when using files for input and output, the errors still occur. 

The input read files look perfect. I've checked them every way that I can think of, and there's nothing that I can find wrong with them. The read files work in all other software that I've tested (eg., fastqc, fastx_toolkit, seqtk). 

I'm using `bowtie2  v2.3.2` installed via conda (as part of humann2). The conda install is: `py27pl5.22.0_1    bioconda`. 

I've been trying to debug this for days and haven't found a solution. I'm about ready to give up on using bowtie2.

**Comment by ch4rr0** [comment]
Would it be possible to upload the sample files used for creating the index and aligning? That will help with recreating this issue.

**Comment by nick-youngblut** [comment]
The bowtie2 index files are part of the `metaphlan2` repo and can found [here](https://bitbucket.org/biobakery/metaphlan2/src/f19f7cfdc990aa9f2d9837a89bcc5188912c8a82/db_v20/?at=default)

**Comment by ch4rr0** [comment]
Thank you. Any chance I can get the sample input file `R1-R2_trim_filt.fq`?

**Comment by nick-youngblut** [comment]
Here's [a link](https://drive.google.com/file/d/0B__aSAgVKwQ6MW5pc05NcGVFcjg/view?usp=sharing) to the read file. I was hoping that the issue was resolved with `bowtie2 v2.3.3.1` (installed from the bioconda channel via conda), but that still generates the same errors. The error I just got on the provided reads with `bowtie2 v2.3.3.1` is: 

```
Argument "\0JJJJJJJJJFJFJJJ%0" isn't numeric in bitwise and (&) at /ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/7ccc9eb0/bin/bowtie2 line 562, <BT> line 1066498.
Traceback (most recent call last):
  File "/ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/7ccc9eb0/bin/metaphlan2.py", line 1164, in <module>
    min_alignment_len = pars['min_alignment_len'])
  File "/ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/7ccc9eb0/bin/metaphlan2.py", line 645, in run_bowtie2
    if o[2][-1] != '*':
IndexError: list index out of range
Argument "" isn't numeric in bitwise and (&) at /ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/7ccc9eb0/bin/bowtie2 line 562, <BT> line 1199301.
Argument "" isn't numeric in bitwise and (&) at /ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/7ccc9eb0/bin/bowtie2 line 562, <BT> line 1214784.
Argument "\0JJJJFJJJA%0" isn't numeric in bitwise and (&) at /ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/7ccc9eb0/bin/bowtie2 line 562, <BT> line 1268687.
Argument "\0J-A<<A<-FFF7FFAF<-)7<<FA7--AFFJ%0" isn't numeric in bitwise and (&) at /ebio/abt3_projects/software/dev/llmgp/.snakemake/conda/7ccc9eb0/bin/bowtie2 line 562, <BT> line 1423039.
```

As far as I can tell, this only happens on the SGE compute cluster, but running the same jobs on a server running the same version of Ubuntu (16.04) doesn't generate these errors.

**Comment by nick-youngblut** [comment]
Just checking: any progress on this? I can provide more read files or details about my SGE compute cluster if needed.

To get around this issue, I'm currently using `snakemake --restart-times 20`. For most samples, `metaphlan2` (using bowtie2) will succeed at least once during these 20 tries, but some samples require even more retries than this. Obviously, this is not very efficient in regards to time and computation, but it's working.

**Comment by ch4rr0** [comment]
I have yet to recreate this issue even with the sample input provided. I will continue trying today on an available cluster.

**Comment by ch4rr0** [comment]
@nick-youngblut -- can you list the precise series of steps, humann2, metaphalan etc., that you took when recreating this issue? I have ran the below command line across several environments without issue. I am beginning to think that this may be an issue with the humann2 pipeline and not bowtie.

`bowtie2 --quiet --no-unal --very-sensitive -S out.sam -mpa_v20_m200 -p 30 -U R1-R2_trim_filt.fq`

**Comment by nick-youngblut** [comment]
Thanks so much for checking! The humann2 pipeline just uses metaphlan2, so it would have to be some issue there. The python code in metaphlan2 for calling bowtie2 is:

```python
def run_bowtie2(  fna_in, outfmt6_out, bowtie2_db, preset, nproc, 
                  file_format = "multifasta", exe = None, 
                  samout = None,
                  min_alignment_len = None,
                  ):

    read_fastx = "read_fastx.py"
    try:
        subprocess.check_call( [read_fastx,"-h"] )
    except Exception as e:
        try:
            read_fastx =os.path.join((os.path.join(os.path.dirname(__file__),"utils")),read_fastx)
        except Exception as e:
            sys.stderr.write( "OSError: fatal error running \'read_fastx.py\'. Is it in the system path?\n" )
            sys.exit(1)

    try:
        if fna_in: 
            readin = subp.Popen([read_fastx,fna_in],stdout=subp.PIPE)
        else: 
            readin = subp.Popen([read_fastx],stdin=sys.stdin,stdout=subp.PIPE)
        bowtie2_cmd = [ exe if exe else 'bowtie2', 
                        "--quiet", "--no-unal", 
                        "--"+preset,
                        "-S","-",
                        "-x", bowtie2_db,
                         ] + ([] if int(nproc) < 2 else ["-p",str(nproc)])
        bowtie2_cmd += ["-U", "-"] # if not stat.S_ISFIFO(os.stat(fna_in).st_mode) else []
        bowtie2_cmd += (["-f"] if file_format == "multifasta" else []) 
        p = subp.Popen( bowtie2_cmd, stdout=subp.PIPE, stdin=readin.stdout ) 
        readin.stdout.close()
        lmybytes, outf = (mybytes,bz2.BZ2File(outfmt6_out, "w")) if outfmt6_out.endswith(".bz2") else (str,open( outfmt6_out, "w" ))
        
        try:
            if samout:
                if samout[-4:] == '.bz2':
                    sam_file = bz2.BZ2File(samout, 'w')
                else:
                    sam_file = open(samout, 'w')
        except IOError:
            sys.stderr.write( "IOError: Unable to open sam output file.\n" )
            sys.exit(1)

        for line in p.stdout:
            if samout:
                sam_file.write(line)
            if line[0] != '@':
                o = read_and_split_line(line)
                if o[2][-1] != '*':
                    if min_alignment_len == None\
                        or max([int(x.strip('M')) for x in\
                                re.findall(r'(\d*M)', o[5])]) >= min_alignment_len:
                        outf.write( lmybytes("\t".join([o[0],o[2]]) +"\n") )
        
        outf.close()
        if samout:
            sam_file.close()
        p.communicate()

    except OSError:
        sys.stderr.write( "OSError: fatal error running BowTie2. Is BowTie2 in the system path?\n" )
        sys.exit(1)
    except ValueError:
        sys.stderr.write( "ValueError: fatal error running BowTie2.\n" )
        sys.exit(1)
    except IOError:
        sys.stderr.write( "IOError: fatal error running BowTie2.\n" )
        sys.exit(1)
    if p.returncode == 13:
        sys.stderr.write( "Permission Denied Error: fatal error running BowTie2." 
          "Is the BowTie2 file in the path with execution and read permissions?\n" )
        sys.exit(1)
    elif p.returncode != 0:
        sys.stderr.write("Error while running bowtie2.\n")
        sys.exit(1)
```


I tried modifying to write to a file instead of streaming to see if that would help:

```python
# modified version of `run_bowtie2`
def run_bowtie2(  fna_in, outfmt6_out, bowtie2_db, preset, nproc,
                  file_format = "multifasta", exe = None,
                  samout = None,
                  min_alignment_len = None, tmp_dir = '.'):

    # temporary output from bowtie
    sam_tmp_file = tf.NamedTemporaryFile(dir=os.path.split(fna_in)[0]).name

    try:
        bowtie2_cmd = [ exe if exe else 'bowtie2',
                        "--quiet",
                        "--no-unal",
                        "--" + preset,
                        "-S", sam_tmp_file,
                        "-x", bowtie2_db,
                         ] + ([] if int(nproc) < 2 else ["-p",str(nproc)])
        bowtie2_cmd += ["-U", fna_in]
        bowtie2_cmd += (["-f"] if file_format == "multifasta" else [])
        bowtie2_cmd = ' '.join(bowtie2_cmd)
        sys.stderr.write('Calling bowtie2: {}\n'.format(bowtie2_cmd))
        subp.call(bowtie2_cmd, shell=True)
        lmybytes, outf = (mybytes,bz2.BZ2File(outfmt6_out, "w")) if outfmt6_out.endswith(".bz2") else (str,open( outfmt6_out, "w" ))

        try:
            if samout:
                if samout[-4:] == '.bz2':
                    sam_file = bz2.BZ2File(samout, 'w')
                else:
                    sam_file = open(samout, 'w')
        except IOError:
            sys.stderr.write( "IOError: Unable to open sam output file.\n" )
            sys.exit(1)

        with open(sam_tmp_file) as inF:
            for line in inF:
                if samout:
                    sam_file.write(line)
                if line[0] != '@':
                    o = read_and_split_line(line)
                    if o[2][-1] != '*':
                        if min_alignment_len == None\
                            or max([int(x.strip('M')) for x in\
                                re.findall(r'(\d*M)', o[5])]) >= min_alignment_len:
                            outf.write( lmybytes("\t".join([o[0],o[2]]) +"\n") )
        outf.close()
        if samout:
            sam_file.close()
        try:
            os.remove(sam_tmp_file)
        except OSError:
            pass

    except OSError:
        sys.stderr.write( "OSError: fatal error running BowTie2. Is BowTie2 in the system path?\n" )
        sys.exit(1)
    except ValueError:
        sys.stderr.write( "ValueError: fatal error running BowTie2.\n" )
        sys.exit(1)
    except IOError:
        sys.stderr.write( "IOError: fatal error running BowTie2.\n" )
        sys.exit(1)
```

However, that didn't help. Do you see anything weird/problematic about the bowtie2 call in metaphlan2? I'll try running bowtie2 by itself on my compute cluster to see if I'm still having the same issues.

**Comment by dnbaker** [comment]
Can you modify the script to write the temporary files which you feed into bowtie to disk and inspect them yourself? Trying to open them with another parser like pysam, bwa, or PicardTools (especially PicardTools) could validate your format. I suspect that the pipeline is generating invalid intermediate fastq files.

**Comment by nick-youngblut** [comment]
Sorry for the slow reply. I've done some testing with bowtie2 writing out a SAM file. Inspecting the SAM files shows that some lines are partially over-written with what should be the next line. It appears that multiple threads are trying to write to the same file and they are over-writing each other's output. When I use the "--reorder" flag, this issue seems to go away, suggesting that it is indeed a result of multiple threads over-writing each other (I'm assuming that only 1 thread is used to write the output then using "--reorder") 

This only seems to happen when I run bowtie2 on a SGE compute cluster running Ubuntu 16.04.3. The cluster uses a NFS file system, but the same file server is also used when running bowtie2 on a headless server, and there's no over-writing issue there. 

I'm fine with using the "--reorder" flag to deal with the problem. The challenge is that I'm using bioinformatic tools that call bowtie2, and so I would have to alter that code so that the "--reorder" flag is used.

**Comment by BenLangmead** [comment]
Note that we have a potential fix for this issue. If you have the chance, there is an alpha build with a potential fix linked to from issue #158. If you can try it and let us know if the problem is fixed, we'd appreciate it. Either way, a new release with that fix is coming very soon.

**Comment by nick-youngblut** [comment]
Thanks for working on a fix! I'll give it a try but I've found it hard to consistently recreate the error without running large datasets that take a while to process. Regardless, I'll try it out.

**Comment by tomas-pluskal** [comment]
Hi, 
I am seeing the same error message during our Trinity runs on a LSF computing cluster running Ubuntu 14.04. We are using bowtie2-2.3.3.1-linux-x86_64 in 2 threads. Just letting you know.

**Comment by conchoecia** [comment]
I am getting the same error with reads that I renamed with headers in this format: `>1_1/1` `>1_2/1` `>1_3/1`

This error is coming up while using Trinity v2.6.6 in Docker.

```
Argument ">1_258980/1%0ATGGATCTAAGCAACCTATTGAAAATGTTTGATTAGCAAATTC..." isn't numeric in bitwise and (&) at /usr/local/bin/bowtie2 line 562, <BT> line 909667.
Argument ">1_258981/1%0AATTTCGTACATCTAGCCGTGTTTTTGCTATAGCCTACCAAGT..." isn't numeric in bitwise and (&) at /usr/local/bin/bowtie2 line 562, <BT> line 909669.
Argument ">1_258982/1%0AACAGCCACACCTGAATGTTGTATTTGCGCTTAGAAAAATGGC..." isn't numeric in bitwise and (&) at /usr/local/bin/bowtie2 line 562, <BT> line 909671.
```

**Comment by ch4rr0** [comment]
We think this issue was most likely due to a multi-threaded bug that we have since fixed in v2.3.4 and above. I will close this thread for now, but please feel free to reopen if this issue arises in said versions.

**Comment by mikemartinez99** [comment]
Hello, I am incorporating Bowtie2 as a rule in a Snakemake workflow. I have Bowtie2 installed through a conda environment (2.3.5.1) and the rule was working fine yesterday, but now today I am receiving the same error mentioned in the closed issue. 

When I test my command outside of Snakemake, it works fine, but I am really stumped since it was working yesterday. 

I've attached my command below:

`rule tRNA_align:
    input:
        trim_1 = "trimming/{sample}.R1.trim.fastq.gz",
    output:
        #sam = "alignment/{sample}.aln.sam",
        alignLog = "tRNA_alignment/{sample}.alignment.log.txt",
        unalign = "tRNA_unaligned/{sample}.unalign.fastq",
        srtBam = "tRNA_alignment/{sample}.srt.bam"
    conda: "trax_env"
    resources: cpus="10", maxtime="6:00:00", mem_mb="60gb"
    params:
        sample = lambda wildcards: wildcards.sample,
        bt2_index = config["bt2_index"],
        maxMaps = config["maxMaps"],
        nPenalty = config["nPenalty"],
    shell: """
    
        #----- Run Bowtie2 alignment (removed the --reorder arg from original code to speed up...don't see the need with SE data)
        bowtie2 \
            -x {params.bt2_index} \
            -U {input.trim_1} \
            -D 20 -R 3 -N 1 -L 12 -i S,1,0.50 \
            -k {params.maxMaps} \
            --very-sensitive \
            --np {params.nPenalty} \
            --ignore-qual \
            --un {output.unalign} \
            -p {resources.cpus} \
            --verbose \
            -S tRNA_alignment/{params.sample}.aln.sam 2> {output.alignLog}

        #----- subset reads for aligned length > 15 & < 90bp 
        samtools view -h tRNA_alignment/{params.sample}.aln.sam | \
            awk 'BEGIN {{OFS="\t"}} $1 ~ /^@/ || ((length($10) > 15 && length($10) <= 90))' | \
            samtools view -Sb - > tRNA_alignment/{params.sample}.bam

        #----- filter for any reads with MAPQ <=1
        samtools view -h -q 2 tRNA_alignment/{params.sample}.bam > tRNA_alignment/{params.sample}.sub.bam

        #----- Sort and filter the bam file
        samtools sort -@ 4 tRNA_alignment/{params.sample}.sub.bam > {output.srtBam}
        samtools index {output.srtBam}

        #----- Remove temp files
        rm -rf tRNA_alignment/{params.sample}.aln.sam
        rm -rf tRNA_alignment/{params.sample}.bam
        rm -rf tRNA_alignment/{params.sample}.sub.bam
    """`

The error I am receiving is as follows: 
Argument "  trimming/IB4.R1.trim.fastq.g" isn't numeric in bitwise and (&) at /dartfs-hpc/rc/lab/G/GMBSR_bioinfo/misc/owen/sharedconda/miniconda/envs/trax_env/bin/bowtie2 line 590, <BT> line 3.
Argument "Output file: "" isn't numeric in bitwise and (&) at /dartfs-hpc/rc/lab/G/GMBSR_bioinfo/misc/owen/sharedconda/miniconda/envs/trax_env/bin/bowtie2 line 590, <BT> line 5.
Argument "Sanity checking: disable" isn't numeric in bitwise and (&) at /dartfs-hpc/rc/lab/G/GMBSR_bioinfo/misc/owen/sharedconda/miniconda/envs/trax_env/bin/bowtie2 line 590, <BT> line 7.

It looks like for some reason file paths are being truncated, spaces are being added, and the output I specified with -S is getting ignored completely. It keeps getting hung up at the sam file stage. 

I have checked that all paths specified in my config file are correct, I have also re-indexed my the bowtie index and verified that it is valid.

Any help would be greatly appreciated! I've also tried implementing the solution from Nick Youngblut (removing the --reorder flag which was originally in my command) but this did not work either. I also tried upgrading to version 2.5.4, also to no avail.
